exchanges/okx/ratelimit.go

```diff
@@ -0,0 +1,938 @@
+	getGeeksRate                          = 10
+
+	// Sub Account Endpoints
+
+	viewSubaccountListRate                             = 2
+	getSubaccountTradingBalanceRate                    = 2
+	getSubaccountFundingBalanceRate                    = 2
+	historyOfSubaccountTransferRate                    = 6
+	masterAccountsManageTransfersBetweenSubaccountRate = 1
+	setPermissingOfTransferOutRate                     = 1
+	getCustoryTradingSubaccountListRate                = 1
+	gridTradingRate                                    = 20
+	amendGridAlgoOrderRate                             = 20
+	stopGridAlgoOrderRate                              = 20
+	getGridAlgoOrderListRate                           = 20
+	getGridAlgoOrderHistoryRate                        = 20
+	getGridAlgoOrderDetailsRate                        = 20
+	getGridAlgoSubOrdersRate                           = 20
+	getGridAlgoOrderPositionsRate                      = 20
+	spotGridWithdrawIncomeRate                         = 20
+
+	// Market Data
+
+	getTickersRate               = 20
+	getIndexTickersRate          = 20
+	getOrderBookRate             = 20
+	getCandlesticksRate          = 40
+	getCandlesticksHistoryRate   = 20
+	getIndexCandlesticksRate     = 20
+	getMarkPriceCandlesticksRate = 20
+	getTradesRequestRate         = 20
+	getTradesHistoryRate         = 10
+	get24HTotalVolumeRate        = 2
+	getOracleRate                = 1
+	getExchangeRateRequestRate   = 1
+	getINdexComponentsRate       = 20
+	getBlockTickersRate          = 20
+	getBlockTradesRate           = 20
+
+	// Public Data Endpoints
+
+	getInstrumentsRate                         = 20
+	getDeliveryExerciseHistoryRate             = 40
+	getOpenInterestRate                        = 20
+	getFundingRate                             = 20
+	getFundingRateHistoryRate                  = 20
+	getLimitPriceRate                          = 20
+	getOptionMarketDateRate                    = 20
+	getEstimatedDeliveryExercisePriiceRate     = 10
+	getDIscountRateAndInterestFreeQuotaRate    = 2
+	getSystemTimeRate                          = 10
+	getLiquidationOrdersRate                   = 2
+	getMarkPriceRate                           = 10
+	getPositionTiersRate                       = 10
+	getInterestRateAndLoanQuotaRate            = 2
+	getInterestRateAndLoanQuoteForVIPLoansRate = 2
+	getUnderlyingRate                          = 20
+	getInsuranceFundRate                       = 10
+	unitConvertRate                            = 10
+
+	// Trading Data Endpoints
+
+	getSupportCoinRate                    = 5
+	getTakerVolumeRate                    = 5
+	getMarginLendingRatioRate             = 5
+	getLongShortRatioRate                 = 5
+	getContractsOpeninterestAndVolumeRate = 5
+	getOptionsOpenInterestAndVolumeRate   = 5
+	getPutCallRatioRate                   = 5
+	getOpenInterestAndVolumeRate          = 5
+	getopenInterestAndVolumeRate          = 5
+	getTakerFlowRate                      = 5
+
+	// Status Endpoints
+
```

Comment:
earn sub catagory not here as well:
- [ ] https://www.okx.com/docs/en/#rest-api-earn-get-offers
- [ ] https://www.okx.com/docs/en/#rest-api-earn-purchase
- [ ] https://www.okx.com/docs/en/#rest-api-earn-redeem
- [ ] https://www.okx.com/docs/en/#rest-api-earn-cancel-purchases-redemptions
- [ ] https://www.okx.com/docs/en/#rest-api-earn-get-active-orders
- [ ] https://www.okx.com/docs/en/#rest-api-earn-get-order-history